src/lib.rs

```diff
@@ -21,6 +21,8 @@ extern crate base64;
 extern crate bytes;
 #[macro_use] extern crate futures;
 extern crate futures_cpupool;
+#[cfg(feature = "compat")]
+extern crate http as http_types;
```

Comment:
I'd eventually move the `hyper::http` module to be `hyper::proto`, most likely. That'd allow using the `http` namespace for the crate. I realize fixing all the renames is tedious, so I'm not saying it's required, I can get to it eventually :)

Reply:
I'm happy todo that as a smaller PR if you don't mind!